### PR TITLE
Slit profile

### PR DESCRIPTION
**Pull request by rcooke-ast**
New functionality included to determine the spatial slit profile from a trace frame (i.e. a flat field frame with the same slit length as the science frame).

### CODE REVIEW

File: doc/tracing.rst [hunk 1]
```diff
@@ -91,4 +91,40 @@ slits, by giving the following keyword argument::
...
+of parameters that determine the fit properties
+are given by::
+
+    reduce slitprofile perform True
+    reduce slitprofile method bspline
+    reduce slitprofile params [n]
+
+where n in the last line should be an integer or
+floating point number.
+The first line can be used to turn off the
```
Comment (profxj) [inline doc/tracing.rst]: Is this now the default?  It isn't clear.

File: pypit/artrace.py [hunk 2]
```diff
@@ -2380,3 +2379,61 @@ def slit_image(slf, det, scitrace, obj, tilts=None):
...
+    -------
+    slit_profile : ndarray
+      An image containing the slit profile
+    """
+    from pypit import arcytrace
+
+    nslits = slf._lordloc[det - 1].shape[1]
+    slit_profiles = np.zeros_like(mstrace)
+    # Set the number of ticks in the spectral direction
+    ntcky = 20  # Default value
```
Comment (profxj) [inline pypit/artrace.py]: maybe define in the def above?

**Comment by profxj** [comment]
Looks good.  Merge if you are happy with it.

**Comment by rcooke-ast** [comment]
First, I'll generate some fancy QA... stay tuned.

**Comment by rcooke-ast** [comment]
Now with fancy QA, and a few fixes to the echelle tilt tracing algorithm, and updated APF Levy settings... Merging...

**Pull request merged**